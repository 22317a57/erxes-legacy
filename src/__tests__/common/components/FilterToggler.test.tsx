import { mount, shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import FilterToggler from '../../../modules/inbox/components/leftSidebar/FilterToggler';

describe('FilterToggler component', () => {
  const defaultProps = {
    groupText: 'Channels',
    isOpen: false,

    toggle: (params: { isOpen: boolean }) => null
  };

  test('renders FilterToggler successfully', () => {
    shallow(<FilterToggler {...defaultProps} />);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<FilterToggler {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });

  test('renders with default props', () => {
    const wrapper = mount(<FilterToggler {...defaultProps} />);
    const props = wrapper.props();

    expect(props).toMatchObject(defaultProps);
  });
});
