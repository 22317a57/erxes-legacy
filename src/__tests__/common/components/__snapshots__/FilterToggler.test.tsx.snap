// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterToggler component snapshot matches 1`] = `
<div
  className="sc-bNQFlB jSwnvB"
  onClick={[Function]}
>
  Channels
  <i
    className="icon-downarrow sc-bwzfXH hpxbGw"
    icon="downarrow"
  />
</div>
`;
